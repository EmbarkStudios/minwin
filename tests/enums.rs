mod utils;
use utils::*;

// Ensures we can emit some but not all of a simple enum's values
test!(
    simple,
    [
        BindConfig::Minwin(Default::default()),
        BindConfig::Minwin(MinwinBindConfig {
            enum_style: EnumStyle::Minwin,
            ..Default::default()
        }),
        BindConfig::Minwin(MinwinBindConfig {
            enum_style: EnumStyle::Minwin,
            use_rust_casing: true,
            ..Default::default()
        }),
        BindConfig::Bindgen,
    ],
    [
        "Windows.Win32.Foundation.GENERIC_READ",
        "Windows.Win32.Foundation.GENERIC_WRITE",
        "Windows.Win32.Foundation.GENERIC_ALL",
        "Windows.Win32.Storage.FileSystem.FILE_TYPE_DISK",
        "Windows.Win32.Storage.FileSystem.GetFileType",
        "Windows.Win32.Storage.FileSystem.ClfsMgmtLogWriteNotification",
        "Windows.Win32.Storage.FileSystem.ReadLogNotification",
    ]
);
