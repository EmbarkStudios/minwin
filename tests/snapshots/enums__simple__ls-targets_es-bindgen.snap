---
source: tests/enums.rs
expression: bo.bindings
---
#![allow(non_snake_case, non_upper_case_globals, non_camel_case_types)]
::windows_targets::link!(
    "clfsw32.dll" "system" fn ReadLogNotification(hLog : HANDLE, pNotification : * mut
    CLFS_MGMT_NOTIFICATION, lpOverlapped : * mut OVERLAPPED) -> BOOL
);
::windows_targets::link!(
    "kernel32.dll" "system" fn GetFileType(hFile : HANDLE) -> FILE_TYPE
);
pub type CLFS_MGMT_NOTIFICATION_TYPE = i32;
pub const ClfsMgmtLogWriteNotification: CLFS_MGMT_NOTIFICATION_TYPE = 3;
pub type FILE_TYPE = u32;
pub const FILE_TYPE_DISK: FILE_TYPE = 1;
pub type GENERIC_ACCESS_RIGHTS = u32;
pub const GENERIC_ALL: GENERIC_ACCESS_RIGHTS = 268435456;
pub const GENERIC_WRITE: GENERIC_ACCESS_RIGHTS = 1073741824;
pub const GENERIC_READ: GENERIC_ACCESS_RIGHTS = 2147483648;
pub type BOOL = i32;
#[repr(C)]
pub struct CLFS_MGMT_NOTIFICATION {
    pub Notification: CLFS_MGMT_NOTIFICATION_TYPE,
    pub Lsn: CLS_LSN,
    pub LogIsPinned: u16,
}
#[repr(C)]
pub struct CLS_LSN {
    pub Internal: u64,
}
pub type HANDLE = isize;
#[repr(C)]
pub struct OVERLAPPED {
    pub Internal: usize,
    pub InternalHigh: usize,
    pub Anonymous: OVERLAPPED_0,
    pub hEvent: HANDLE,
}
#[repr(C)]
pub union OVERLAPPED_0 {
    pub Anonymous: ::std::mem::ManuallyDrop<OVERLAPPED_0_0>,
    pub Pointer: *mut ::core::ffi::c_void,
}
#[repr(C)]
pub struct OVERLAPPED_0_0 {
    pub Offset: u32,
    pub OffsetHigh: u32,
}

